src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -240,8 +241,17 @@ protected MessageImpl createMessage(boolean send, String address, MultiMap heade
     completionHandler.handle(Future.succeededFuture());
   }
 
-  protected <T> boolean addLocalRegistration(String address, HandlerRegistration<T> registration,
-                                             boolean replyHandler, boolean localOnly) {
+  private static class LocalRegistrationResult<T> {
+    final HandlerHolder<T> holder;
+    final boolean newAddress;
+    public LocalRegistrationResult(HandlerHolder<T> holder, boolean newAddress) {
+      this.holder = holder;
+      this.newAddress = newAddress;
+    }
+  }
+
+  private <T> LocalRegistrationResult addLocalRegistration(String address, HandlerRegistration<T> registration,
```

Comment:
Should be `private <T> LocalRegistrationResult<T>`

Reply:
will try